Sources/NIOPosix/Bootstrap.swift

```diff
@@ -2108,61 +2116,140 @@ extension NIOPipeBootstrap {
         try await self._takingOwnershipOfDescriptors(
             input: input,
             output: output,
-            channelInitializer: channelInitializer,
-            postRegisterTransformation: { $0.makeSucceededFuture($1) }
+            channelInitializer: channelInitializer
         )
     }
 
+    /// Create the `PipeChannel` with the provided input file descriptors.
+    ///
+    /// The input file descriptor must be distinct.
+    ///
+    /// - Note: If this method returns a succeeded future, SwiftNIO will close `input` when the `Channel`
+    ///         becomes inactive. You _must not_ do any further operations to `input`, including `close`.
+    ///         If this method returns a failed future, you still own the file descriptor and are responsible for
+    ///         closing them.
+    ///
+    /// - Parameters:
+    ///   - input: The _Unix file descriptor_ for the input (ie. the read side).
+    ///   - channelInitializer: A closure to initialize the channel. The return value of this closure is returned from the `connect`
+    ///   method.
+    /// - Returns: The result of the channel initializer.
     @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
-    func _takingOwnershipOfDescriptors<ChannelInitializerResult, PostRegistrationTransformationResult: Sendable>(
+    public func takingOwnershipOfDescriptor<Output: Sendable>(
```

Comment:
`Output` is a bit misleading in light of `input`/`output` descriptors. Should we use "Result" or something super-generic?

Reply:
We are using `Output` in all the other methods some I am hesitant to change it just for this one. Also why do you suggest making this public?

Reply:
> We are using `Output` in all the other methods some I am hesitant to change it just for this one. 

Yeah, I get that, it's probably fine as is.

> Also why do you suggest making this public?

I didn't, _you_ made it public!